qtt/instrument_drivers/virtualAwg/sequencer.py

```diff
@@ -98,17 +99,33 @@ def make_marker(period, uptime=0.2, offset=0.0, repetitions=1, name='marker'):
             Dict: *NAME*, *TYPE*, *WAVE* keys containing values; sequence name,
                   sequence data type and the actual qupulse sequencePT respectively.
         """
-        if uptime <= 0 or offset < 0:
-            raise ValueError('Invalid argument value (uptime <= 0 or offset < 0)!')
-        if uptime + offset > period:
-            raise ValueError('Invalid argument value (uptime + offset > period)!')
-        input_variables = {'period': period * Sequencer.__sec_to_ns,
-                           'uptime': uptime * Sequencer.__sec_to_ns,
-                           'offset': offset * Sequencer.__sec_to_ns}
-        sequence_data = (Templates.marker(name), input_variables)
-        return {'name': name, 'wave': SequencePT(*((sequence_data,)*repetitions)),
+
+        if abs(offset) > period:
+            raise ValueError('Invalid argument value for offset: |{}| > {}!'.format(offset, period))
+        if not 0 < uptime < period:
+            raise ValueError("Invalid argument value for uptime '{}'!".format(uptime))
+        input_variables = Sequencer.__calculate_pulse_parameters(offset, period, uptime)
+        sequence_data = (Templates.rollover_marker(name), input_variables)
+        return {'name': name, 'wave': SequencePT(*((sequence_data,) * repetitions)),
                 'type': DataTypes.QU_PULSE, 'uptime': uptime, 'offset': offset}
 
+    @staticmethod
+    def __calculate_pulse_parameters(offset, period, uptime):
```

Comment:
The name is not descriptive enough. The method is really about the marker parameters and not a general pulse. Maybe `_calculate_marker_pulse_parameters`?